docs/script.rst

```diff
@@ -212,52 +208,53 @@ not broken up by `new line` character(s)::
+Thus, you can define a chunk of code and then pass it around as if it were a string or an integer.
 
-More formally, you can create functions that are defined `first class objects`.
+More formally, you can create functions that are defined as `first class objects`.
 
 ::
 
     square = { it * it }
 
 
 The curly brackets around the expression ``it * it`` tells the script interpreter to treat this expression as code.
-In this case, the designator ``it`` refers to whatever value is given to the function. Then this compiled function is
-assigned to the variable `'square`` much like those above. So now we can do something like this::
+In this case, the designator ``it`` refers to whatever value is passed to the function when it is called. This compiled function is
+assigned to the variable ``square``, much like variable assignments shown previously. Now we can do something like this::
 
     println square(9)
 
 and get the value 81.
 
 
-This is not very interesting until we find that we can pass this function ``square`` around as a method argument.
-There are some built in functions that take a function like this as an argument. One example is the ``collect`` method on lists.
-For example::
+This is not very interesting until we find that we can pass the function ``square`` as an argument to other functions or methods.
+Some built-in functions take a function like this as an argument. One example is the ``collect`` method on lists::
 
     [ 1, 2, 3, 4 ].collect(square)
 
 
-This expression says, create an array with the values 1,2,3 and 4, then call the `collect` method, passing in the
-closure we defined above. The collect method runs through each item in the array, calls the closure on the item,
+This expression says: Create an array with the values 1, 2, 3 and 4, then call its ``collect`` method, passing in the
+closure we defined above. The ``collect`` method runs through each item in the array, calls the closure on the item,
 then puts the result in a new array, resulting in::
 
     [ 1, 4, 9, 16 ]
 
 
-For more methods that you can call with closures as arguments, see the `Groovy GDK documentation <http://groovy.codehaus.org/groovy-jdk/>`_.
+For more methods that you can call with closures as arguments, see the `Groovy GDK documentation <http://docs.groovy-lang.org/latest/html/groovy-jdk/>`_.
 
 
-By default closures take 1 parameter called ``it``, you can also create closures with named parameters.
-For example the method ``Map.each()`` can take a closure with two variables, to which it binds the `key` and the associated `value`::
+By default, closures take a single parameter called ``it``, but you can also create closures with multiple, custom-named parameters.
+For example, the method ``Map.each()`` can take a closure with two arguments, to which it binds the `key` and the associated `value`
+for each key-value pair in the ``Map``. Here, we use the obvious variable names ``key`` and ``value`` in our closure::
```

Comment:
I tried to make it clearer here that the actual variable names, `key` and `value` are conventional, but ultimately arbitrary, by calling them "custom-named parameters" and "the obvious variable names". I'm new to Groovy, but take it that `it` is _not_ arbitrary -- it's the name that the language uses.